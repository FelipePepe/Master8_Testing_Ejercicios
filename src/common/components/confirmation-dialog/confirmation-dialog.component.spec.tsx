import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ConfirmationDialogComponent } from './confirmation-dialog.component';
import { Typography } from '@material-ui/core';

describe('Confirmation Dialog Components specs', () => {
  it('Should display confirmation dialog components click on Accept', () => {
    // Arrange
    const props = {
      isOpen: true,
      title: 'test title',
      labels: {
        closeButton: 'test Close',
        acceptButton: 'test Accept',
      },
      onAccept: jest.fn(),
      onClose: jest.fn(),
    };

    const itemName = 'Hello Testing World';

    // Act
    render(
      <ConfirmationDialogComponent {...props}>
        <Typography variant="body1">{itemName}</Typography>
      </ConfirmationDialogComponent>
    );

    const element = screen.getByRole('button', {
      name: props.labels.acceptButton,
    });

    userEvent.click(element);

    // Assert
    expect(props.onAccept).toHaveBeenCalled();
  });
  it('Should display confirmation dialog components click on Accept', () => {
    // Arrange
    const props = {
      isOpen: true,
      title: 'test title',
      labels: {
        closeButton: 'test Close',
        acceptButton: 'test Accept',
      },
      onAccept: jest.fn(),
      onClose: jest.fn(),
    };

    const itemName = 'Hello Testing World';

    // Act
    render(
      <ConfirmationDialogComponent {...props}>
        <Typography variant="body1">{itemName}</Typography>
      </ConfirmationDialogComponent>
    );

    const element = screen.getByRole('button', {
      name: props.labels.acceptButton,
    });

    userEvent.click(element);

    // Assert
    expect(props.onAccept).toHaveBeenCalled();
  });

  it('Should display confirmation dialog components', () => {
    // Arrange
    const props = {
      isOpen: true,
      title: 'test title',
      labels: {
        closeButton: 'test Close',
        acceptButton: 'test Accept',
      },
      onAccept: jest.fn(),
      onClose: jest.fn(),
    };

    const itemName = 'Hello Testing World';

    // Act
    render(
      <ConfirmationDialogComponent {...props}>
        <Typography variant="body1">{itemName}</Typography>
      </ConfirmationDialogComponent>
    );

    const element = screen.getByRole('presentation');

    // Assert
    expect(element).toBeInTheDocument();
  });

  it('Should display confirmation dialog components using snapshot testing', () => {
    // Arrange
    const props = {
      isOpen: true,
      title: 'test title',
      labels: {
        closeButton: 'test Close',
        acceptButton: 'test Accept',
      },
      onAccept: jest.fn(),
      onClose: jest.fn(),
    };

    const itemName = 'Hello Testing World';

    // Act
    render(
      <ConfirmationDialogComponent {...props}>
        <Typography variant="body1">{itemName}</Typography>
      </ConfirmationDialogComponent>
    );

    const element = screen.getByRole('presentation');

    // Assert
    expect(element).toMatchSnapshot();
  });
});
