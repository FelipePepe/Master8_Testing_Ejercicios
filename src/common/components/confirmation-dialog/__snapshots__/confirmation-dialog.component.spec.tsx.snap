// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Dialog Components specs Should be displayed confirmation dialog components using snapshot testing 1`] = `
<div
  class="MuiDialog-root"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <div
        class="MuiDialogTitle-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6"
        >
          test title
        </h2>
      </div>
      <div
        class="MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hello Testing World
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            test Close
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            test Accept
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Confirmation Dialog Components specs Should display confirmation dialog components using snapshot testing 1`] = `
<div
  class="MuiDialog-root"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <div
        class="MuiDialogTitle-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6"
        >
          test title
        </h2>
      </div>
      <div
        class="MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hello Testing World
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            test Close
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            test Accept
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
